### PR TITLE
Feature #17 sign up buttons validation

### LINKED ISSUE #20: Create users and crete documetn data
# REQUIREMENTS
Register new users
create a documetn to users data
install firestore module

# Acceptance Criteria
List all required acceptance criteria
- [ ] The user can log with email and password 
- [ ] when the user log firestore genered a new document

### PR DESCRIPTION
**Pull request by HRuvalcabaP**
# Description
1: We created a function to created new users.
2: We install firestore and genered a document for store flights data.
3: We created a new document inside users collection with userId with loged userId

 # Testing
No.
 
  ## Screenshots 
image

### CODE REVIEW

**Review by Joaquin-Niembro — APPROVED** [review]

**Review by Ing-Heber — APPROVED** [review]
Well done!, as a tip keep in mind create a variable to store the colors(mentors will ask you to do that sooner or later) and I would create a class name for the img in order to make your code cleaner. Nice job!

**Review by JordiGN — APPROVED** [review]
Good job, nice and clean, just as my teammate said, I would suggest creating a script for the colors, you will need it later

File: flightreservation/Views/SignUp.js [hunk 1]
```diff
@@ -1,35 +1,167 @@
...
+import auth from '@react-native-firebase/auth';
+
+import firestore from '@react-native-firebase/firestore';
+
+const SignUp = ({ navigation }) => {
+  const [firstNameInput, setFirstNameInput] = useState(false);
+  const [emailInput, setEmailInput] = useState(false);
+  const [passwordInput, setPasswordInput] = useState(false);
+  const [showPassword, setShowPassword] = useState(true);
+  const [toggleCheckBox1, setToggleCheckBox1] = useState(false);
```
Comment (develiasdaniel) [inline flightreservation/Views/SignUp.js:16]: I recommend that its components be more descriptive

File: flightreservation/Views/SignUp.js [hunk 2]
```diff
@@ -1,35 +1,167 @@
...
+  const [toggleCheckBox2, setToggleCheckBox2] = useState(false);
```
Comment (develiasdaniel) [inline flightreservation/Views/SignUp.js:17]: I recommend that its components be more descriptive

File: flightreservation/Views/SignUp.js [hunk 3]
```diff
@@ -1,35 +1,167 @@
...
+              <Icon
+                style={styles.eye}
+                name="eye"
+                color={passwordInput ? '#5C6EF8' : '#A9A9A9'}
+                size={25}
+                onPress={() => setShowPassword(!showPassword)}
+              />
+            </View>
+            <Text style={styles.textFeedback}>
+              Use 8 or more characters with a mix of letter, numbers and symbols.
```
Comment (develiasdaniel) [inline flightreservation/Views/SignUp.js:66]: as my partner said, you better have a file of constants for both colors and texts

File: flightreservation/Views/SignUp.js [hunk 4]
```diff
@@ -1,35 +1,167 @@
...
+                        setErrorEmail('That email address is already in use!');
+                      }
+                      if (error.code === 'auth/invalid-email') {
+                        setErrorEmail('The email address is badly formatted!');
+                      }
+                      if (error.code === 'auth/weak-password') {
+                        setErrorPassword('Password should be at least 6 characters');
+                      }
+                      console.error(error);
+                    });
```
Comment (develiasdaniel) [inline flightreservation/Views/SignUp.js:122]: I think it is better to put it in a function outside the return

**Review by develiasdaniel — APPROVED** [review]
Excellent work!, improve for the following 🚀

File: flightreservation/Views/SignUp.js [hunk 5]
```diff
@@ -1,35 +1,167 @@
...
+            />
+            <Text style={styles.labelCheck}>Subscribe for select product updates.</Text>
+          </View>
+          <View style={styles.containerSpacing}>
+            <View style={styles.shadow}>
+              <TouchableOpacity
+                style={allValid ? styles.buttonEnabled : styles.buttonDisabled}
+                disabled={!allValid}
+                onPress={() => {
+                  auth()
```
Comment (develiasdaniel) [inline flightreservation/Views/SignUp.js:100]: I think it is better to put it in a function outside the return

File: flightreservation/Views/SignUp.js [hunk 6]
```diff
@@ -1,35 +1,167 @@
...
+                      }
+                      if (error.code === 'auth/invalid-email') {
+                        setErrorEmail('The email address is badly formatted!');
+                      }
+                      if (error.code === 'auth/weak-password') {
+                        setErrorPassword('Password should be at least 6 characters');
+                      }
+                      console.error(error);
+                    });
+                }}
```
Comment (develiasdaniel) [inline flightreservation/Views/SignUp.js:123]: I think it is better to put it in a function outside the return

File: flightreservation/Views/SignUp.js (hunk 5 above)
Comment (develiasdaniel) [inline flightreservation/Views/SignUp.js:100]: I think it is better to put "auth()..." in a function outside the return

**Review by JacintoOrtiz15745 — APPROVED** [review]

**Review by jjcr11 — APPROVED** [review]

**Pull request merged**